src/pages/docs/guides/chat/export-chat.mdx

```diff
@@ -0,0 +1,132 @@
+---
+title: "Guide: Export chat data to your own systems"
+meta_description: "Learn how to export chat data from Ably Chat to your own systems."
+meta_keywords: "chat, data, export, stream, storage, Ably, chat SDK, realtime messaging, dependability, cost optimization"
+---
+
+Ably Chat is a simple and easy-to-use realtime chat solution that handles any scale from 1:1 and small group chats to large livestream chats with millions of users.
+
+Ably holds data for the purpose of providing realtime experiences. While Ably Chat provides flexible data retention for messages (30 days by default, up to a year on request), some applications may need longer-term storage or additional control over their data.
+
+This guide presents different ways to use Ably Chat and store chat data in your own systems, which can help you meet your data retention requirements, as well as help you build more complex use cases such as search, analytics, and more.
+
+## Different ways to export data from Ably Chat
+
+This guide covers the following options to export data from Ably Chat:
+
+1. Using [outbound webhooks](/docs/platform/integrations/webhooks). [HTTP endpoint](/docs/platform/integrations/webhooks/generic), [AWS Lambda](/docs/platform/integrations/webhooks/lambda), and others.
+2. Using [outbound streaming](/docs/platform/integrations/streaming). Stream to your own [Kafka](/docs/platform/integrations/streaming/kafka), [Kinesis](/docs/platform/integrations/streaming/kinesis), and others.
+3. Using an [Ably queue](/docs/platform/integrations/queues).
+4. Publishing via your own servers.
+5. Using the Chat History endpoint.
+
+## Decoding and storing messages
+
+Regardless of the delivery mechanism, you will need to decode the received messages into Chat messages. Details of the mapping from Ably Pub/Sub messages to Chat messages are available in the [Integrations](/docs/chat/integrations) documentation.
+
+After performing the decoding to get your chat `Message` object, you can:
+
+1. Save it to your own database. You can index by `serial`, this is the global unique identifier for a message, and is also used to sort messages in the canonical global order.
+2. If the message already exists by `serial`, it means you have received an update, delete, or reaction summary update. To check if you need to update the message, you can use the `version.serial` to compare the latest version of the message you have received with the version of the message you have in your database. Lexicographically higher means newer version.
+3. If you want to store reaction summaries, always update the reactions field when receiving a reaction summary update (action `4` or `message.summary`).
+
+<Code>
+```javascript
+const saveOrUpdateMessage = (message) => {
+  // Check if the message already exists in your own database by `serial`
+  const existingMessage = await getMessageBySerial(message.serial);
```

Comment:
Reflecting on this example - for many database engines this would be racy and may throw errors (e.g. if a message and an update happen in very quick succession). Whether or not to update would usually happen atomically at the database layer e.g. SQL's `INSERT ... ON DUPLICATE KEY UPDATE`

Reply:
I can see how people might be tempted to just add implementations for `getMessageBySerial`, `saveMessage` and `updateMessage` instead of actually writing a `saveOrUpdateMessage` for their own system.

That wasn't the purpose of the example when I wrote it. I meant this example to show the logic: how should you decide insert, update or discard.

I'm not sure how to fix it:

1. explain in the text that this is an example to illustrate how to compare versions/etc, not meant to be a code template
2. change it to use SQL and some assumed schema (I can mention said schema early in the article)? That might not be great for those who don't use SQL but I guess it'll be readable enough to be understood by anyone.
3. something else?